identity/src/main/java/com/stripe/android/identity/ui/ConsentScreen.kt

```diff
@@ -0,0 +1,285 @@
+package com.stripe.android.identity.ui
+
+import android.graphics.Typeface
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.CircularProgressIndicator
+import androidx.compose.material.Divider
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.semantics.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.viewinterop.AndroidView
+import com.google.android.material.composethemeadapter.MdcTheme
+import com.stripe.android.identity.R
+import com.stripe.android.identity.networking.Resource
+import com.stripe.android.identity.networking.Status
+import com.stripe.android.identity.networking.models.VerificationPage
+import com.stripe.android.identity.networking.models.VerificationPage.Companion.isUnsupportedClient
+import com.stripe.android.identity.networking.models.VerificationPage.Companion.requireSelfie
+import com.stripe.android.identity.utils.setHtmlString
+
+@Composable
+internal fun ConsentScreen(
+    verificationState: Resource<VerificationPage>,
+    onMerchantViewCreated: (ImageView) -> Unit,
+    onSuccess: (VerificationPage) -> Unit,
+    onFallbackUrl: (String) -> Unit,
+    onError: (Throwable) -> Unit,
+    onConsentAgreed: (Boolean) -> Unit,
+    onConsentDeclined: (Boolean) -> Unit
+) {
+    MdcTheme {
+        when (verificationState.status) {
+            Status.SUCCESS -> {
+                requireNotNull(verificationState.data).let { verificationPage ->
+                    LaunchedEffect(Unit) {
+                        onSuccess(verificationPage)
+                    }
+
+                    if (verificationPage.isUnsupportedClient()) {
+                        onFallbackUrl(verificationPage.fallbackUrl)
+                    } else {
```

Comment:
This would also trigger on every `verificationState` change right? I think this should be 

```
LaunchedEffect(verificationPage.unsupportedClient) {
    if(unsupported == true)...
}

Reply:
Yes, I think there are two ways of writing it, chosen the 2nd way because it's less verbose, wdyt?
1.
```
LaunchedEffect(verificationPage.unsupportedClient) {
    if (verificationPage.unsupportedClient) {
        onFallbackUrl(verificationPage.fallbackUrl)
    }
}
if (!verificationPage.unsupportedClient) {
    SuccessUI(
        verificationPage,
        onMerchantViewCreated,
        onConsentAgreed,
        onConsentDeclined
    )
}
```

2.
```
if (verificationPage.isUnsupportedClient()) {
    LaunchedEffect(Unit) {
        onFallbackUrl(verificationPage.fallbackUrl)
    }
} else {
    SuccessUI(
        verificationPage,
        onMerchantViewCreated,
        onConsentAgreed,
        onConsentDeclined
    )
}
```

Reply:
in this case we should be 👌  since the computation of `if (verificationPage.unsupportedClient)` is extremely simple, but in general 1 (using launchedEffect with a key) is a better approach to prevent the logic from running if the condition remains constant.

In cases like animations compositions can trigger many tines and even small computations can cause dropped frames, but definitely no impact here : )